### DISCUSSION TITLE
How to config AdventureLog with Nginx Proxy Manager?

### DISCUSSION THREAD

**Discussion by Aloe-recite** [seanmorley15/AdventureLog#361] [Q&A] [answered]
Hi all,
I run a proxmox server with a container running NPM and another container running AdventureLog (both came from Tteck's scripts, BTW - I have been the user asking for Adventurelog btw!).
I can access adventure log without issues if I use IP:PORT, but if I use the address "https://adventurelog.mydomain.it" ad configure in NPM (only available locally) I get a 403 error in console and cannot login.
Can someone please patiently point out what I need to do in order to make things work? I tried editing the .env file located in adventure log/frontend and substituting the 192.xxx IP with the domain above, but nothing changed.
Please be patient since I'm new to all this :)
Thanks!

**Comment by seanmorley15** [comment]
Hi @Aloe-recite! I would check the `ORIGIN` variable for the frontend. Make sure its set to the value of where you access your instance, for you https://adventurelog.mydomain.it
I hope this helps, keep me updated!

**Reply by Aloe-recite** [reply]
Hi Sean, I have this in the /frontend/.env file:
```
PUBLIC_SERVER_URL=https://advlog.mydomain.tld
BODY_SIZE_LIMIT=Infinity
ORIGIN='https://advlog.mydomain.tld'
```
the address is indeed correct. HTTPS is served by Nginx proxy manager using a wildcard certificate.
Did you mean this variable and this file, or something else?
thank you for your help

**Reply by seanmorley15** [reply]
Hi @Aloe-recite, I would try this as your env file
```
PUBLIC_SERVER_URL=http://server:8000
BODY_SIZE_LIMIT=Infinity
ORIGIN=https://advlog.mydomain.tld # What you access the app from in your browser
```
The public server URL can use `server:8000` assuming the services are in the same compose file stack.
Give this a shot and keep me updated!

**Reply by Aloe-recite** [reply]
@seanmorley15 bad news unfortunately, this makes the login apparently work but then I end up in a bank page (only the login, signup and searcher are visible) with error 500 in console :(
I tried the following:
```
PUBLIC_SERVER_URL=http://192.168.1.225:8000
BODY_SIZE_LIMIT=Infinity
ORIGIN=https://advlog.mydomain.tld
```
and with this I can login, but...all images are lost. I get a plethora of error like this:
[Warning] [blocked] The page at https://advlog.domain.tld/worldtravel requested insecure content from http://192.168.1.225:8000/media/flags/ye.png. This content was blocked and must be served over HTTPS. (scheduler.DU2ID3VB.js, line 1)
Any idea?

**Reply by seanmorley15** [reply, marked as answer]
@Aloe-recite, I would check two variables, `PUBLIC_SERVER_URL` on the frontend and `PUBLIC_URL` on the backend. I would try `PUBLIC_SERVER_URL=http://server:8000`, because of the internal docker networks. Also for the PUBLIC_URL, it seems like you are using some kind of proxy for the app and not for the ngnix media service, this is causing the HTTPS warning. I would make sure that you are also proxying that and using a domain like media.mydomain.tld and using it as the `PUBLIC_URL`. If that does not work and the login and stuff does, don't worry too much because the update I am releasing this Wednesday removes the need for a separate nginx container and just includes it in the server container.  This will prevent scenarios like this form happening! Sorry for the confusion and I hope this helps 🤞

**Reply by Aloe-recite** [reply]
Thanks Sean, no big deal using ip:port if I cannot make it work my way; my capabilities are just a notch above "non-exixtent" so that's to be expected. Hopes up for the new upcoming version 🤞🏻 I suppose that many others want AdventureLog under a personal domain / subdomain and many use NPM (or Caddy or Traefik, these three together should cover the vast majority of use cases) so maybe it could be worth documenting in detail how to achieve this :)
thank you for your effort and keep p the good work - I'll keep you updated after the next release

**Reply by Aloe-recite** [reply]
@seanmorley15 thanks Sean, that did it! I used Nginx Proxy Manager to proxy "adventurelogmedia.mydomain.tld" to the container IP:800=, then put https://adventurelogmedia.mydomain.tld as PUBLIC_SERVER_URL on the frontend and PUBLIC_URL on the backend. Seems to login and work for now, I'll try creating a few adventure images to check later on! I suggest you document the process somewhere, there are many NPM users in the wild :D

**Reply by seanmorley15** [reply]
@Aloe-recite, I am super glad its working well now! This will be added to the docs website under the new "guides" section. Thanks so much for your patience!
Happy travels!

**Reply by Freddster** [reply]
@seanmorley15, as someone who used the Proxmox installation from ```bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/adventurelog.sh)"```, the guide you created unfortunately does not cater very much to this audience due to mainly referencing the Docker environment and how it is set up with the ```docker-compose.yml```. As such, I found this discussion thread more useful, and I believe it would be useful to have a section dedicated to the approach if you start with Proxmox.

For future people finding this post:
Create two proxies, one for the frontend and one for the backend (choose your own names).
Frontend: https://adventurelog.mydomain.tld
Backend: https://adventurelogmedia.mydomain.tld

Update the frontend configuration and ensure that ```PUBLIC_SERVER_URL``` points to the backend proxy and ```ORIGIN``` points to the frontend proxy.
```
PUBLIC_SERVER_URL=https://adventurelogmedia.mydomain.tld
ORIGIN='https://adventurelog.mydomain.tld'
```

For the backend configuration file, update the ```PUBLIC_URL``` to point to the backend proxy and the ```FRONTEND_URL``` to point to the frontend proxy.
```
PUBLIC_URL='https://adventurelogmedia.mydomain.tld'
FRONTEND_URL='https://adventurelog.mydomain.tld'
```

In both files, unchanged lines have been omitted from the snippets.

Additionally, in the backend configuration, you can add the frontend and backend proxies to the ```CSRF_TRUSTED_ORIGINS```setting. However, whether this is a recommended practice, I will leave to someone more experienced to answer.

**Reply by peetezaah** [reply]
>As such, I found this discussion thread more useful, and I believe it would be useful to have a section dedicated to the approach if you start with Proxmox.  

Totally agree and thank you for compiling the necessary steps! Started homelabbing very recently and this issue took a bit more effort to find an answer to.  
For completeness, I would only add the ports for each proxy: frontend is "3000" and backend is "8000", as well as the location of each file: frontend is `/opt/adventurelog/frontend/.env` and backend is at `/opt/adventurelog/backend/server/.env`.  
Have you ever submitted an issue to the [community-scripts repo](https://github.com/community-scripts/ProxmoxVE/issues)? I was thinking of creating one, so they change the warning box to something more complete, with the information you summarized, or at least a link to your post.

**Reply by Freddster** [reply]
> Have you ever submitted an issue to the [community-scripts repo](https://github.com/community-scripts/ProxmoxVE/issues)? I was thinking of creating one, so they change the warning box to something more complete, with the information you summarized, or at least a link to your post.

I have not created an issue, either with [community-scripts repo](https://github.com/community-scripts/ProxmoxVE/issues) or with AdventureLog.

**Reply by hotswapster** [reply]
@Freddster thanks for your pointers with the Proxmox setup. I also have the same running but I still get the error. In your nginx config, did you proxy to ports 3000 (frontend) and 8000 (backend)?

I have setup the domain names and fronend and backend confs as you suggested but still get 500: Internal error.

**Reply by Gu4n** [reply]
Same issue here. Followed @Freddster's instructions but like @hotswapster I wind up with `500: Internal error` when logging in to the frontend. I have no issues accessing public Collections or logging into the backend over the Nginx Proxy.

**Reply by hotswapster** [reply]
@Gu4n I worked it out. I had the following two lines in my nginx config, and when I removed them it started working with no error.

```
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
```

**Reply by hotswapster** [reply]
> @Gu4n I worked it out. I had the following two lines in my nginx config, and when I removed them it started working with no error.
> 
> ```
>         proxy_set_header Upgrade $http_upgrade;
>         proxy_set_header Connection "upgrade";
> ```

I stand corrected - Images still don't work. But collection and adventure creation is fine
